cpp/src/arrow/python/iterators.h

```diff
@@ -27,10 +27,19 @@ namespace arrow {
 namespace py {
 namespace internal {
 
-// Visit the Python sequence, calling the given callable on each element.
-// If the callable returns a non-OK status, iteration stops and the status is returned.
-template <class UnaryFunction>
-inline Status VisitSequence(PyObject* obj, UnaryFunction&& func) {
+// Visit the Python sequence, calling the given callable on each element.  If
+// the callable returns a non-OK status, iteration stops and the status is
+// returned.
+//
+// The call signature for Visitor must be Visit(PyObject*, bool*
```

Comment:
Apparently the visitor also takes an index argument.

Reply:
fixed